core/services/workflows/syncer/handler.go

```diff
@@ -135,12 +158,119 @@ func (h *eventHandler) Handle(ctx context.Context, event WorkflowRegistryEvent)
 }
 
 // workflowRegisteredEvent handles the WorkflowRegisteredEvent event type.
-// TODO: Implement this method
 func (h *eventHandler) workflowRegisteredEvent(
-	_ context.Context,
-	_ WorkflowRegistryEvent,
+	ctx context.Context,
+	event WorkflowRegistryEvent,
 ) error {
-	return ErrNotImplemented
+	payload, ok := event.Data.(WorkflowRegistryWorkflowRegisteredV1)
+	if !ok {
+		return fmt.Errorf("invalid data type %T for event", event.Data)
+	}
+
+	wfID := hex.EncodeToString(payload.WorkflowID[:])
+
+	cma := h.emitter.With(
+		platform.KeyWorkflowID, wfID,
+		platform.KeyWorkflowName, payload.WorkflowName,
+		platform.KeyWorkflowOwner, hex.EncodeToString(payload.WorkflowOwner),
+	)
+
+	// Download the contents of binaryURL, configURL and secretsURL and cache them locally.
+	binary, err := h.fetcher(ctx, payload.BinaryURL)
+	if err != nil {
+		logCustMsg(ctx, cma, fmt.Sprintf("failed to fetch binary: %v", err), h.lggr)
```

Comment:
Is there a way to put the custom logging up one level? That way handlers can just return an error, and the thing calling the handler handles the error

One hitch that might make this difficult is the fact that your emitter is presumably event specific.